### PR TITLE
ReadStream that replaces invalid XML chars RSRVR-110

### PR DESCRIPTION
**Pull request by adamdickmeiss**

### CODE REVIEW

File: util/src/main/java/org/folio/reservoir/util/readstream/XmlFixerMapper.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    if (pending == null) {
+      input = buffer;
+    } else {
+      pending.appendBuffer(buffer);
+      input = pending;
+    }
+    tail = 0;
+    for (front = 0; front < input.length(); front++) {
+      byte leadingByte = input.getByte(front);
+      if (leadingByte < 32 && leadingByte != 9 && leadingByte != 10 && leadingByte != 13) {
```
Comment (jakub-id) [inline util/src/main/java/org/folio/reservoir/util/readstream/XmlFixerMapper.java:37]: We don't need to care about surrogate blocks, right?

**Review by jakub-id — APPROVED** [review]

File: util/src/main/java/org/folio/reservoir/util/readstream/XmlFixerMapper.java (hunk 1 above)
Comment (adamdickmeiss) [reply util/src/main/java/org/folio/reservoir/util/readstream/XmlFixerMapper.java:37]: I dont know what you mean

File: server/src/main/java/org/folio/reservoir/server/OaiPmhClientService.java [hunk 2]
```diff
@@ -592,7 +593,7 @@ private Future<Void> listRecordsResponse(Storage storage, OaiPmhStatus job,
     if (res.statusCode() != 200) {
       return handleBadResponse(res);
     }
-    XmlParser xmlParser = XmlParser.newParser(res);
+    XmlParser xmlParser = XmlParser.newParser(new XmlFixer(res));
```
[the remark this answers is not in the document]
Comment (adamdickmeiss) [reply server/src/main/java/org/folio/reservoir/server/OaiPmhClientService.java:596]: Yes

File: util/src/main/java/org/folio/reservoir/util/readstream/XmlFixerMapper.java (hunk 1 above)
Comment (adamdickmeiss) [reply util/src/main/java/org/folio/reservoir/util/readstream/XmlFixerMapper.java:37]: These will never be part of a multi byte sequence.  Multibyte are all >= 128.

**Pull request merged**